runelite-client/src/main/java/net/runelite/client/plugins/statusbars/StatusBarsOverlay.java

```diff
@@ -27,77 +27,149 @@
 import net.runelite.client.ui.overlay.OverlayPosition;
-import net.runelite.client.ui.overlay.components.TextComponent;
 import net.runelite.client.util.ImageUtil;
 
 class StatusBarsOverlay extends Overlay
 {
 	private static final Color PRAYER_COLOR = new Color(50, 200, 200, 175);
 	private static final Color QUICK_PRAYER_COLOR = new Color(57, 255, 186, 225);
-	private static final Color BACKGROUND = new Color(0, 0, 0, 150);
 	private static final Color HEALTH_COLOR = new Color(225, 35, 0, 125);
 	private static final Color POISONED_COLOR = new Color(0, 145, 0, 150);
 	private static final Color VENOMED_COLOR = new Color(0, 65, 0, 150);
 	private static final Color HEAL_COLOR = new Color(255, 112, 6, 150);
 	private static final Color PRAYER_HEAL_COLOR = new Color(57, 255, 186, 75);
-	private static final Color OVERHEAL_COLOR = new Color(216, 255, 139, 150);
+	private static final Color ENERGY_HEAL_COLOR = new Color (199,  118, 0, 218);
+	private static final Color RUN_ENERGY_COLOR = new Color(199, 174, 0, 220);
+	private static final Color SPECIAL_ATTACK_COLOR = new Color(3, 153, 0, 195);
 	private static final int HEIGHT = 252;
 	private static final int RESIZED_BOTTOM_HEIGHT = 272;
-	private static final int WIDTH = 20;
-	private static final int PADDING = 1;
 	private static final int IMAGE_SIZE = 17;
-	private static final int HEALTH_LOCATION_X = 0;
-	private static final int PRAYER_LOCATION_X = 1;
 	private static final int RESIZED_BOTTOM_OFFSET_Y = 12;
 	private static final int RESIZED_BOTTOM_OFFSET_X = 10;
-	private static final int OVERHEAL_OFFSET = 2;
-	private static final int HEAL_OFFSET = 3;
-	private static final int ICON_AND_COUNTER_OFFSET_X = 1;
-	private static final int ICON_AND_COUNTER_OFFSET_Y = 21;
-	private static final int SKILL_ICON_HEIGHT = 35;
-	private static final int COUNTER_ICON_HEIGHT = 18;
-	private static final int OFFSET = 2;
+	private static final int MAX_SPECIAL_ATTACK_VALUE = 100;
+	private static final int MAX_RUN_ENERGY_VALUE = 100;
+	static final Color BACKGROUND = new Color(0, 0, 0, 150);
+	static final Color OVERHEAL_COLOR = new Color(216, 255, 139, 150);
+	static final int OVERHEAL_OFFSET = 2;
+	static final int HEAL_OFFSET = 3;
+	static final int ICON_AND_COUNTER_OFFSET_X = 1;
+	static final int ICON_AND_COUNTER_OFFSET_Y = 21;
+	static final int SKILL_ICON_HEIGHT = 35;
+	static final int COUNTER_ICON_HEIGHT = 18;
+	static final int OFFSET = 2;
+	static final int WIDTH = 20;
+	static final int PADDING = 1;
 
 	private final Client client;
 	private final StatusBarsConfig config;
 	private final SkillIconManager skillIconManager;
-	private final TextComponent textComponent = new TextComponent();
+	private final SpriteManager spriteManager;
 	private final ItemStatChangesService itemStatService;
 
 	private final Image prayerImage;
+	private Image runEnergyImage;
+	private Image specialImage;
+
+	private final Map<BarMode, BarRenderer> barRenderers = new HashMap<>();
```

Comment:
Use an EnumMap here.

```suggestion
	private final Map<BarMode, BarRenderer> barRenderers = new EnumMap<>(BarMode.class);
```